src/sentry/taskworker/consumers/factory.py

```diff
@@ -0,0 +1,140 @@
+from __future__ import annotations
+
+import logging
+import time
+from collections.abc import Mapping, MutableSequence
+
+from arroyo.backends.kafka.consumer import KafkaPayload
+from arroyo.processing.strategies import (
+    CommitOffsets,
+    ProcessingStrategy,
+    ProcessingStrategyFactory,
+    Reduce,
+    RunTask,
+)
+from arroyo.processing.strategies.abstract import MessageRejected
+from arroyo.types import BaseValue, Commit, Message, Partition
+from google.protobuf.timestamp_pb2 import Timestamp
+from sentry_protos.sentry.v1alpha.taskworker_pb2 import (
+    TASK_ACTIVATION_STATUS_PENDING,
+    InflightActivation,
+    TaskActivation,
+)
+
+from sentry.taskworker.inflight_activation_store import InflightTaskStoreSqlite
+
+logger = logging.getLogger("sentry.taskworker.consumer")
+
+
+class TaskWorkerStrategyFactory(ProcessingStrategyFactory[KafkaPayload]):
+    def __init__(
+        self,
+        max_batch_size: int,
+        max_batch_time: int,
+        num_processes: int,
+        max_pending_timeout: int,
+        max_inflight_activation_in_store: int,
+        input_block_size: int | None,
+        output_block_size: int | None,
+    ) -> None:
+        super().__init__()
+        self.max_batch_size = max_batch_size
+        self.max_batch_time = max_batch_time
+        self.input_block_size = input_block_size
+        self.output_block_size = output_block_size
+
+        # Maximum amount of seconds that tasks are allowed to live in the inflight activation store.
+        # After this time elapses, tasks are be deadlettered if they are followed by completed records.
+        self.max_pending_timeout = max_pending_timeout
+
+        # Maximum number of pending inflight activations in the store before backpressure is emitted
+        self.max_inflight_activation_in_store = max_inflight_activation_in_store
+
+        self.inflight_task_store = InflightTaskStoreSqlite()
+
+    def create_with_partitions(
+        self, commit: Commit, _: Mapping[Partition, int]
+    ) -> ProcessingStrategy[KafkaPayload]:
+        def process_message(message: Message[KafkaPayload]) -> InflightActivation:
+            activation = TaskActivation()
+            activation.ParseFromString(message.payload.value)
+            ((_partition, offset),) = message.committable.items()
+            deadletter_at = int(time.time()) + self.max_pending_timeout
```

Comment:
Does using `max_pending_timeout` make sense for calculating `deadletter_at` here?